lib/Runtime/Library/SameValueComparer.h

```diff
@@ -31,39 +31,56 @@ namespace Js
             }
         }
 
+        static hash_t HashDouble(double d)
+        {
+            if (JavascriptNumber::IsNan(d))
+            {
+                return 0;
+            }
+
+            if (zero)
+            {
+                // SameValueZero treats -0 and +0 the same, so normalize to get same hash code
+                if (JavascriptNumber::IsNegZero(d))
+                {
+                    d = 0.0;
+                }
+            }
+
+            __int64 v = *(__int64*)&d;
```

Comment:
We should also not be duplicating the raw double hash function algorithm.  Instead we should call `DefaultComparer<double>` to do that work.  I'll update the PR after I convince myself that the NaN special case is or is not necessary.